translate.js

```diff
@@ -30,14 +30,33 @@ const helper = (xs) => ([x, position, y], val) =>
 const cornersHelper = helper(corners);
 const edgesHelper = helper(edges);
 
-const colorHelper = (colors, color, shade) => {
+const colorHelper = (colors, color, shade, opacityHelper) => {
   const base = colors[color];
-  return (
+  const colorValue = (
     base &&
     (typeof base === 'string' ? base : shade ? base[shade] : base['default'])
   );
+  return opacityHelper ? hexToRgb(colorValue, opacityHelper) : colorValue;
 };
 
+// Adopted from `react-colorful` here: https://github.com/omgovich/react-colorful/blob/master/src/utils/convert.ts
+const hexToRgb = (color, opacityHelper) => {
+  if (!/^#[0-9a-fA-F]{8}$|#[0-9a-fA-F]{6}$|#[0-9a-fA-F]{4}$|#[0-9a-fA-F]{3}$/i.test(color)) return undefined;
```

Comment:
Here's something else I was thinking about: returning `undefined` as I do here triggers the "No translation for..." message. Is there a better alternative, or maybe this is something that should be added?

If a users passes `#FF`, it does need to fail, but I'm not sure if "No translation for..." makes the reason clear enough.

Reply:
Humm, good question. We could add an extra warning to to the console if you think that people won't figure out they have messed up a hex code in their theme.

The alternative is just to translate it without any validation (padding missing hex values with zeros and trimming off excess perhaps) and then the color is going to look "off" which might prompt the developer to go check its value in the theme.

Your call here, I'm happy with either, we can always change/add this at a later date.

Reply:
I did think about no-error translations, either with padding or choosing something like hotpink to really get the point across that something isn't right, though I suppose the worry there is the 1% of times where that does in fact look right. 

I think maybe I'll write up a warning PR in the future. For now there is feedback that can at least point to the problem area for users to check. Better messages can be something for another PR so that this doesn't get too bloated.

Reply:
Perfect 👍 I concur.